cql3/statements/select_statement.cc

```diff
@@ -1320,6 +1325,135 @@ indexed_table_select_statement::find_index_clustering_rows(service::storage_prox
+    schema_ptr schema,
+    uint32_t bound_terms,
+    lw_shared_ptr<const select_statement::parameters> parameters,
+    ::shared_ptr<selection::selection> selection,
+    ::shared_ptr<restrictions::statement_restrictions> restrictions,
+    ::shared_ptr<std::vector<size_t>> group_by_cell_indices,
+    bool is_reversed,
+    forwarding_select_statement::ordering_comparator_type ordering_comparator,
+    std::optional<expr::expression> limit,
+    std::optional<expr::expression> per_partition_limit,
+    cql_stats& stats,
+    std::unique_ptr<cql3::attributes> attrs
+) {
+    return ::make_shared<cql3::statements::forwarding_select_statement>(
+        schema,
+        bound_terms,
+        std::move(parameters),
+        std::move(selection),
+        std::move(restrictions),
+        std::move(group_by_cell_indices),
+        is_reversed,
+        std::move(ordering_comparator),
+        std::move(limit),
+        std::move(per_partition_limit),
+        stats,
+        std::move(attrs)
+    );
+}
+
+forwarding_select_statement::forwarding_select_statement(
+    schema_ptr schema,
+    uint32_t bound_terms,
+    lw_shared_ptr<const forwarding_select_statement::parameters> parameters,
+    ::shared_ptr<selection::selection> selection,
+    ::shared_ptr<restrictions::statement_restrictions> restrictions,
+    ::shared_ptr<std::vector<size_t>> group_by_cell_indices,
+    bool is_reversed,
+    forwarding_select_statement::ordering_comparator_type ordering_comparator,
+    std::optional<expr::expression> limit,
+    std::optional<expr::expression> per_partition_limit,
+    cql_stats& stats,
+    std::unique_ptr<cql3::attributes> attrs
+) : select_statement(
+    schema,
+    bound_terms,
+    std::move(parameters),
+    std::move(selection),
+    std::move(restrictions),
+    std::move(group_by_cell_indices),
+    is_reversed,
+    std::move(ordering_comparator),
+    std::move(limit),
+    std::move(per_partition_limit),
+    stats,
+    std::move(attrs)
+) {
+}
+
+future<::shared_ptr<cql_transport::messages::result_message>>
+forwarding_select_statement::do_execute(
+    service::storage_proxy& proxy,
+    service::forward_service& forwarder,
+    service::query_state& state,
+    const query_options& options
+) const {
+    tracing::add_table_name(state.get_trace_state(), keyspace(), column_family());
+
+    auto cl = options.get_consistency();
+    validate_for_read(cl);
+
+    auto now = gc_clock::now();
+
+    const source_selector src_sel = state.get_client_state().is_internal()
+            ? source_selector::INTERNAL : source_selector::USER;
+    ++_stats.query_cnt(src_sel, _ks_sel, cond_selector::NO_CONDITIONS, statement_type::SELECT);
+
+    _stats.select_bypass_caches += _parameters->bypass_cache();
+    _stats.select_allow_filtering += _parameters->allow_filtering();
+    _stats.select_partition_range_scan += _range_scan;
+    _stats.select_partition_range_scan_no_bypass_cache += _range_scan_no_bypass_cache;
```

Comment:
I think it deserves a new counter. This will get incremented once for a select statement that scans the entire table, whereas normal select statements will increment once per page. For a query that has huge impact on the cluster, it is invisible in the metrics!

Reply:
These all look like old counters.

Reply:
Ah, there's a new one below.